### ISSUE TITLE
Does not start epoch.

### ISSUE THREAD

**Issue by AndreCostaWind** [matterport/Mask_RCNN#1781]
I need help. Please.
Even after long hours with 4 GPUs v100 does not start epoch ...
Does not display errors.
I tested with color images. Grayscale images ..
I am based on the balloon example.
I currently have 8 class. Plus the background.
I've done some successful training with just one class. Now with I am having this problem.
Any idea?
Thanks.
<img src="http://www.wind.net.br/print.png">

**Comment by noamnav** [comment]
Same here.  Posted [this issue](https://github.com/matterport/Mask_RCNN/issues/1783) yesterday, have a look.

**Comment by AndreCostaWind** [comment]
If anyone has the same problem ..
I found the origin of my problem.
The training got stuck in the dataset validation step.
Because a annotation tool bug it occurred that one of the xml files got dots beyond the image area.
Summing up. If you have any problems with the dataset this crash occurs.

**Comment by AndreCostaWind** [comment]
Excuse me.
Now that I saw your answer.
I found my problem.
The lock is in dataset validation.
In one of the images the mask point has dropped to the edge of the image exactly.
And that created the problem.

And I still fumbled with the screen and saved in a new issue :)

**Comment by noamnav** [comment]
@AndreCostaWind can you please elaborate?
I am unsuccessfully trying to run the shapes sample - don't believe there is bug in the train set as its machine generated?

**Comment by AndreCostaWind** [comment]
OK…

I noticed that epoch didn't start because there were problems with the dataset.

Today I have everything very customized.
Both the training script and the xml pattern I use for annotations.
I use a program of my own to make annotations as well.

It turned out that in this particular dataset the objects highlighting the masks were too big and my tool couldn't make it. I had to resort to a free program.

My training script is pretty much the ballons example, which I customized.

https://engineering.matterport.com/splash-of-color-instance-segmentation-with-mask-r-cnn-and-tensorflow-7c761e238b46

I train with multiple classes and more than one object in each image.

In my case, I use xml to make annotations.
I started with the Pascal Voc 2012 standard. But today I made some data inclusions.

How I found the problem ...
I made a script to check them all files, image and xml.
First we need to test if there are any structure errors in xml. Parse and see if it loads correctly.

Need to test if required xml nodes do not return empty or null.

And it's really worth checking if the maximum and minimum points of the X and Y coordinates are within the image size limit. This is what caused my problem.

Another detail. I've had problems with previous datasets using images with different sizes and resolutions.
Some networks accept well. Others not.
Just in case. Today I standardize the size and resolution of the all images.
